File: 백준/Silver/1475. 방 번호/방 번호.py
rnum=input()
num_list=list(rnum)
count_list=[]
for i in range(len(num_list)):
    if num_list[i]=='9':
        num_list[i]='6'
six=num_list.count('6')
if six%2==0:
    count_list.append(six/2)
else:
    count_list.append(six//2+1)
for i in range(len(num_list)):
    if num_list[i]!='6':
        count_list.append(num_list.count(num_list[i]))
print(int(max(count_list)))

File: 백준/Silver/1475. 방 번호/방 번호.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String rnum = scanner.nextLine();
        char[] num_list = rnum.toCharArray();
        int[] count_list = new int[10];
        
        for (int i = 0; i < num_list.length; i++) {
            if (num_list[i] == '9') {
                num_list[i] = '6';
            }
        }
        
        int six = 0;
        for (char c : num_list) {
            if (c == '6') {
                six++;
            }
        }
        
        if (six % 2 == 0) {
            count_list[6] = six / 2;
        } else {
            count_list[6] = six / 2 + 1;
        }
        
        for (char c : num_list) {
            if (c != '6') {
                count_list[Character.getNumericValue(c)]++;
            }
        }
        
        int maxCount = 0;
        for (int count : count_list) {
            maxCount = Math.max(maxCount, count);
        }
        
        System.out.println(maxCount);
    }
}
